aiidalab_widgets_base/computational_resources.py

```diff
@@ -1196,16 +1196,26 @@ def on_setup_code(self, _=None):
                 "append_text",
             ]
 
+            containerized_codes_additional_items = [
+                "image_name",
+                "engine_command",
+            ]
+
             kwargs = {key: getattr(self, key).value for key in items_to_configure}
 
+            # Check for additional keys needed for orm.ContainerizedCode
+            for container_key in containerized_codes_additional_items:
+                if container_key in self.code_setup.keys():
+                    kwargs[container_key] = self.code_setup[container_key]
+
             # set computer from its widget value the UUID of the computer.
             computer = orm.load_computer(self.computer.value)
 
             # Checking if the code with this name already exists
             qb = orm.QueryBuilder()
             qb.append(orm.Computer, filters={"uuid": computer.uuid}, tag="computer")
             qb.append(
-                orm.InstalledCode,
+                (orm.InstalledCode, orm.ContainerizedCode),
```

Comment:
```suggestion
                orm.AbstractCode,
```

Reply:
@unkcpz  thanks also for this, indeed it is better to provide the abstract class here